File: jokercsi/Programmers/추억 점수.py
'''
추억 점수 (인물의 그리움 점수)
'''
def solution(name, yearning, photo):
    answer = []
    dict = {}
    for i, v in enumerate(name):
        dict[v] = yearning[i]
    #print(dict)
    
    for i in photo:
        sum = 0
        for j in i:
            try:
                sum += dict[j]
            except:
                print(j)
        answer.append(sum)
    return answer

File: hall_of_fame/llddang/추억점수.cpp
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    
    int idx, semiAnswer;
    for(auto vec : photo){
        semiAnswer = 0;
        for(auto str : vec){
            for(idx = 0; idx < name.size(); idx++){
                if(name[idx] == str){
                    semiAnswer += yearning[idx];
                    break;
                }
            }
        }
        answer.push_back(semiAnswer);
    }
    
    return answer;
}
